cmd/cloud-controller-manager/app/options/options_test.go

```diff
@@ -63,55 +65,90 @@ func TestAddFlags(t *testing.T) {
 		"--profiling=false",
 		"--node-status-update-frequency=10m",
 		"--route-reconciliation-period=30s",
+		"--secure-port=10001",
 		"--min-resync-period=100m",
 		"--use-service-account-credentials=false",
-		"--cert-dir=/a/b/c",
-		"--bind-address=192.168.4.21",
-		"--secure-port=10001",
 	}
 	f.Parse(args)
 
 	expected := &CloudControllerManagerOptions{
-		Generic: cmoptions.GenericControllerManagerOptions{
-			ComponentConfig: componentconfig.KubeControllerManagerConfiguration{
-				CloudProvider:                                   "gce",
-				CloudConfigFile:                                 "/cloud-config",
-				Port:                                            10253,     // Note: InsecureServingOptions.ApplyTo will write the flag value back into the component config
-				Address:                                         "0.0.0.0", // Note: InsecureServingOptions.ApplyTo will write the flag value back into the component config
-				ConcurrentEndpointSyncs:                         5,
-				ConcurrentRSSyncs:                               5,
-				ConcurrentResourceQuotaSyncs:                    5,
-				ConcurrentDeploymentSyncs:                       5,
-				ConcurrentDaemonSetSyncs:                        2,
-				ConcurrentJobSyncs:                              5,
-				ConcurrentNamespaceSyncs:                        10,
-				ConcurrentSATokenSyncs:                          5,
-				ConcurrentServiceSyncs:                          1,
-				ConcurrentGCSyncs:                               20,
-				ConcurrentRCSyncs:                               5,
-				MinResyncPeriod:                                 metav1.Duration{Duration: 100 * time.Minute},
-				NodeMonitorPeriod:                               metav1.Duration{Duration: 5 * time.Second},
-				ResourceQuotaSyncPeriod:                         metav1.Duration{Duration: 5 * time.Minute},
-				NamespaceSyncPeriod:                             metav1.Duration{Duration: 5 * time.Minute},
-				PVClaimBinderSyncPeriod:                         metav1.Duration{Duration: 15 * time.Second},
+		Generic: &cmoptions.GenericControllerManagerOptions{
```

Comment:
Which of the field in `GenericControllerManagerOptions` are actually used in kube- **and** cloud-controller-manager. In other words, `GenericControllerManagerOptions` does not really look generic.

If we could get rid of the `GenericControllerManagerOptions` layer now, we would also get rid of the strange `Generic.GenericComponent`.